src/hierarchies/priors/fa_prior_model.cc

```diff
@@ -0,0 +1,212 @@
+#include "fa_prior_model.h"
+
+double FAPriorModel::lpdf(const google::protobuf::Message &state_) {
+  // Downcast state
+  auto &state = downcast_state(state_).fa_state();
+  // Proto2Eigen conversion
+  Eigen::VectorXd mu = bayesmix::to_eigen(state.mu());
+  Eigen::VectorXd psi = bayesmix::to_eigen(state.psi());
+  // Eigen::MatrixXd eta = bayesmix::to_eigen(state.eta());
```

Comment:
+1 for commented lines in the whole file

Reply:
I am letting this conversation open as I need to move a function here still commented.